internal/authorization/authorizer.go

```diff
@@ -113,22 +81,23 @@ func (p *Authorizer) IsSecondFactorEnabled() bool {
 }
 
 // GetRequiredLevel retrieve the required level of authorization to access the object.
-func (p *Authorizer) GetRequiredLevel(subject Subject, requestURL url.URL) Level {
+func (p *Authorizer) GetRequiredLevel(subject Subject, object Object) Level {
 	logger := logging.Logger()
-	logger.Tracef("Check authorization of subject %s and url %s.", subject.String(), requestURL.String())
+	logger.Tracef("Check authorization of subject %s and url %s.", subject.String(), object.String())
 
-	matchingRules := selectMatchingRules(p.configuration.Rules, p.configuration.Networks, subject, Object{
-		Domain: requestURL.Hostname(),
-		Path:   requestURL.Path,
-	})
+	rule, err := getFirstMatchingRule(p.configuration.Rules, p.configuration.Networks, subject, object)
 
-	if len(matchingRules) > 0 {
-		return PolicyToLevel(matchingRules[0].Policy)
-	}
+	if err != nil {
+		if err == errNoMatchingRule {
+			logger.Tracef("No matching rule for subject %s and url %s... Applying default policy.", subject.String(), object.String())
+		} else {
+			logger.Warnf("Error occurred matching ACL Rules for subject %s and url %s: %v", subject.String(), object.String(), err)
```

Comment:
Can we simulate this? I'm trying to think of what scenario this could actually occur given the validation that is performed at start up too?
Is this just a safety net?

Reply:
Just a safety net.